src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```
@@ -277,10 +279,27 @@ def acr_login(registry_name, resource_group_name=None, username=None, password=N
                 'Unable to authenticate using AAD or admin login credentials. ' +
                 'Please specify both username and password in non-interactive mode.')
 
-    call(["docker", "login",
-          "--username", username,
-          "--password", password,
-          login_server])
+    use_password_stdin = False
+    try:
+        p = Popen(["docker", "login", "--help"], stdout=PIPE, stderr=PIPE)
+        stdout, _ = p.communicate()
+        if b'--password-stdin' in stdout:
+            use_password_stdin = True
+    except:
+        raise CLIError(docker_not_available)
+
+    if use_password_stdin:
+        p = Popen(["docker", "login",
+                   "--username", username,
+                   "--password-stdin",
+                   login_server], stdin=PIPE)
+        p.communicate(input=password.encode())
+    else:
```

bare except is not allowed